File: eval/benchmarks/jacobi-1d-imper/Jacobi1DImper.java
public class Jacobi1DImper {
    static final int N = 128;
    static final int TSTEPS = 80;
    static long[] a = new long[N];
    static long[] b = new long[N];

    static void initArray() {
        for (int i = 0; i < N; i++) {
            a[i] = (i + 2) / N;
            b[i] = (i + 3) / N;
        }
    }

    static void kernel() {
        for (int t = 0; t < TSTEPS; t++) {
            for (int i = 1; i < N - 1; i++) {
                b[i] = (a[i - 1] + a[i] + a[i + 1]) / 3;
            }
            for (int j = 1; j < N - 1; j++) {
                a[j] = b[j];
            }
        }
    }

    static long checksum() {
        long s = 0;
        for (int i = 0; i < N; i++) {
            s += a[i];
        }
        return s;
    }

    public static void main(String[] args) {
        initArray();
        kernel();
        System.err.println("checksum=" + checksum());
    }
}

File: eval/benchmarks/jacobi-1d-imper/jacobi_1d_imper.rs
const N: usize = 128;
const TSTEPS: usize = 80;

fn init_array(a: &mut [i64; N], b: &mut [i64; N]) {
    for i in 0..N {
        a[i] = (i as i64 + 2) / N as i64;
        b[i] = (i as i64 + 3) / N as i64;
    }
}

fn kernel(a: &mut [i64; N], b: &mut [i64; N]) {
    for _ in 0..TSTEPS {
        for i in 1..N - 1 {
            b[i] = (a[i - 1] + a[i] + a[i + 1]) / 3;
        }
        a[1..(N - 1)].copy_from_slice(&b[1..(N - 1)]);
    }
}

fn checksum(a: &[i64; N]) -> i64 {
    a.iter().copied().sum()
}

fn main() {
    let mut a = [0i64; N];
    let mut b = [0i64; N];
    init_array(&mut a, &mut b);
    kernel(&mut a, &mut b);
    eprintln!("checksum={}", checksum(&a));
}
